### ISSUE TITLE
autocorrect_source test helper does not work if it is called multiple times in one example

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#4653]
`autocorrect_source` does not work if it is called multiple times in one example.
See. https://github.com/bbatsov/rubocop/pull/4648#issuecomment-320999689

--------

## Expected behavior

- Make the helper to work well in this case

or 

- Add internal cop for this problem

## Actual behavior

It does not work.

## Steps to reproduce the problem

Apply the following patch

```diff
diff --git a/spec/rubocop/cop/style/string_literals_spec.rb b/spec/rubocop/cop/style/string_literals_spec.rb
index 562b1765..91d1a81e 100644
--- a/spec/rubocop/cop/style/string_literals_spec.rb
+++ b/spec/rubocop/cop/style/string_literals_spec.rb
@@ -6,6 +6,11 @@ describe RuboCop::Cop::Style::StringLiterals, :config do
   context 'configured with single quotes preferred' do
     let(:cop_config) { { 'EnforcedStyle' => 'single_quotes' } }
 
+    it do
+      expect(autocorrect_source('"foo"')).to eq "'foo'"
+      expect(autocorrect_source('"barbaz"')).to eq "'barbaz'"
+    end
+
     it 'registers offense for double quotes when single quotes suffice' do
       inspect_source(['s = "abc"',
                       'x = "a\\\\b"',
```

Run rspec

```bash
$ bundle exec rspec spec/rubocop/cop/style/string_literals_spec.rb:10
Run options:
  include {:focus=>true, :locations=>{"./spec/rubocop/cop/style/string_literals_spec.rb"=>[10]}}
  exclude {:broken=>#<Proc:./spec/spec_helper.rb:42>}

Randomized with seed 61978

RuboCop::Cop::Style::StringLiterals
  configured with single quotes preferred
    should eq "'barbaz'" (FAILED - 1)

Failures:

  1) RuboCop::Cop::Style::StringLiterals configured with single quotes preferred should eq "'barbaz'"
     Failure/Error: expect(autocorrect_source('"barbaz"')).to eq "'barbaz'"

       expected: "'barbaz'"
            got: "'foo'az\""

       (compared using ==)
     # ./spec/rubocop/cop/style/string_literals_spec.rb:11:in `block (3 levels) in <top (required)>'

Finished in 0.07491 seconds (files took 0.73822 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/rubocop/cop/style/string_literals_spec.rb:9 # RuboCop::Cop::Style::StringLiterals configured with single quotes preferred should eq "'barbaz'"

Randomized with seed 61978
```

## RuboCop version

master ( 4abb67df0aa5b53cb42ceaae899636ef7cae2529 )

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
I believe RuboCop RSpec actually has a "one assertion per it block"-cop, and this illustrates well why that's a good idea. 🙂 I'm sure trying to adopt RuboCop RSpec right now would be a huge undertaking though.

**Comment by backus** [comment]
>  I'm sure trying to adopt RuboCop RSpec right now would be a huge undertaking though.

Luckily RuboCop is infinitely configurable ;) Just `--auto-gen-config`!

**Comment by Drenmi** [comment]
To be replaced with an auto-correct test DSL in #4507.
